syncd/syncd_flex_counter.cpp

```diff
@@ -467,10 +465,33 @@ void FlexCounter::collectCounters(
 
             values.emplace_back(counterName, sai_serialize_attr_value(*meta, queueAttr[i]));
         }
-        // Write counters to DB
+        // Store queue attribute values for update
         std::string queueVidStr = sai_serialize_object_id(queueVid);
+        queueAttrsMap[queueVidStr] = values;
+    }
+}
+
+void FlexCounter::updateCounters(
+        _In_ swss::Table &countersTable,
+        _In_ std::map<std::string, std::vector<swss::FieldValueTuple>> &portStats,
```

Comment:
the variable name should really be portsStats since the key is the port, same for other variable names.